train.py

```diff
@@ -31,15 +34,14 @@
                   config.min_occur_count)
     if not config.debug:
         pickle.dump(vocab, open(config.save_vocab_path, 'wb'))
-    with mx.Context(mx.gpu(0) if 'cuda' in os.environ['PATH'] else mx.cpu()):
+    with (mx.gpu(0) if 'cuda' in os.environ['PATH'] else mx.cpu()):
         parser = BiaffineParser(vocab, config.word_dims, config.tag_dims, config.dropout_emb, config.lstm_layers,
                             config.lstm_hiddens, config.dropout_lstm_input, config.dropout_lstm_hidden,
                             config.mlp_arc_size,
                             config.mlp_rel_size, config.dropout_mlp, config.debug)
-        parser.initialize(force_reinit=True)
         data_loader = DataLoader(config.train_file, config.num_buckets_train, vocab)
         # trainer = dy.AdamTrainer(pc, config.learning_rate, config.beta_1, config.beta_2, config.epsilon)
-        trainer = gluon.Trainer(parser.collect_params(), 'adam', {'learning_rate': config.learning_rate})
+        trainer = gluon.Trainer(parser.model.collect_params(), 'adam', {'learning_rate': config.learning_rate})
```

Comment:
seems that this should have been parser.collect_params(), because otherwise the weights in the outer model is not updating.

Reply:
Yes, that's true. Now it's good:

```
Step #99: Acc: arc 0.67, rel 0.88, overall 0.62, loss 0.758
duration : 28.53
Test on development set
LAS 1.44, UAS 63.82
```